RecoParticleFlow/PFProducer/src/MLPFModel.cc

```diff
@@ -0,0 +1,212 @@
+      const auto& clref = ((const reco::PFBlockElementSuperCluster*)&orig)->superClusterRef();
+      if (clref.isNonnull()) {
+        eta = clref->eta();
+        phi = clref->phi();
+        px = clref->position().x();
+        py = clref->position().y();
+        pz = clref->position().z();
+        energy = clref->energy();
+      }
+    }
+
+    float typ_idx = static_cast<float>(elem_type_encoding.at(orig.type()));
+
+    //Must be the same order as in tf_model.py
+    return std::array<float, NUM_ELEMENT_FEATURES>({{typ_idx,
+                                                     pt,
+                                                     eta,
+                                                     phi,
+                                                     energy,
+                                                     layer,
+                                                     depth,
+                                                     charge,
+                                                     trajpoint,
+                                                     eta_ecal,
+                                                     phi_ecal,
+                                                     eta_hcal,
+                                                     phi_hcal,
+                                                     muon_dt_hits,
+                                                     muon_csc_hits}});
+  }
+
+  float normalize(float in) {
+    if (std::abs(in) > 1e4) {
+      return 0.0;
+    } else if (std::isnan(in)) {
+      return 0.0;
+    }
+    return in;
+  }
+
+  int arg_max(std::vector<float> const& vec) {
+    return static_cast<int>(std::distance(vec.begin(), max_element(vec.begin(), vec.end())));
+  }
+
+  reco::PFCandidate make_candidate(int pred_pid, int pred_charge, float pred_e, float pred_eta, float pred_phi) {
+    pred_phi = angle0to2pi::make0To2pi(pred_phi);
+
+    //currently, set the pT from a massless approximation.
+    //later versions of the model may predict predict both the energy and pT of the particle
+    float pred_pt = pred_e / cosh(pred_eta);
+
+    //set the charge to +1 or -1 for PFCandidates that are charged, according to the sign of the predicted charge
+    reco::PFCandidate::Charge charge = 0;
+    if (pred_pid == 11 || pred_pid == 13 || pred_pid == 211) {
+      charge = pred_charge > 0 ? +1 : -1;
+    }
+
+    math::PtEtaPhiELorentzVectorD p4(pred_pt, pred_eta, pred_phi, pred_e);
+
+    reco::PFCandidate cand(
+        0, math::XYZTLorentzVector(p4.X(), p4.Y(), p4.Z(), p4.E()), reco::PFCandidate::ParticleType(0));
+    cand.setPdgId(pred_pid);
+    cand.setCharge(charge);
+
+    return cand;
+  }
+
+  const std::vector<const reco::PFBlockElement*> get_pfelements(
+      edm::Event const& event, const edm::EDGetTokenT<reco::PFBlockCollection> inputTagBlocks) {
+    auto blocks_handle = event.getHandle(inputTagBlocks);
+    const auto& blocks = *blocks_handle;
```

Comment:
```suggestion
    const auto& blocks = event.get(inputTagBlocks);
```

Reply:
right, this needed to be updated. cd20cc6688d